checkov/terraform/tf_parser.py

```diff
@@ -329,6 +330,58 @@ def parse_hcl_module(
 
         return module, tf_definitions
 
+    def parse_multi_graph_hcl_module(
+        self,
+        source_dir: str,
+        source: str,
+        download_external_modules: bool = False,
+        external_modules_download_path: str = DEFAULT_EXTERNAL_MODULES_DIR,
+        parsing_errors: dict[str, Exception] | None = None,
+        excluded_paths: list[str] | None = None,
+        vars_files: list[str] | None = None,
+        external_modules_content_cache: dict[str, ModuleContent | None] | None = None,
+        create_graph: bool = True,
+    ) -> list[Tuple[Module, Dict[TFDefinitionKey, Dict[str, Any]]]]:
+        tf_definitions = self.parse_directory(
+            directory=source_dir, out_evaluations_context={},
+            out_parsing_errors=parsing_errors if parsing_errors is not None else {},
+            download_external_modules=download_external_modules,
+            external_modules_download_path=external_modules_download_path, excluded_paths=excluded_paths,
+            vars_files=vars_files, external_modules_content_cache=external_modules_content_cache
+        )
+        tf_definitions = clean_parser_types(tf_definitions)
+        tf_definitions = serialize_definitions(tf_definitions)
+
+        dirs_to_definitions: dict[str, list[dict[TFDefinitionKey, dict[str, Any]]]] = {}
+        for tf_definition_key, tf_value in tf_definitions.items():
+            source_module = tf_definition_key.tf_source_modules
+            if source_module is None:
+                # No module - add new entry to dirs_to_definitions with the path as key
+                dir_path = os.path.dirname(tf_definition_key.file_path)
+                if dir_path in dirs_to_definitions:
+                    dirs_to_definitions[dir_path].append({tf_definition_key: tf_value})
+                else:
+                    dirs_to_definitions[dir_path] = [{tf_definition_key: tf_value}]
+            else:
+                # iterate over nested modules while adding directories on the way
+                while source_module is not None:
+                    if source_module.nested_tf_module is None:
+                        dir_path = os.path.dirname(source_module.path)
+                        if dir_path in dirs_to_definitions:
+                            dirs_to_definitions[dir_path].append({tf_definition_key: tf_value})
+                        else:
+                            dirs_to_definitions[dir_path] = [{tf_definition_key: tf_value}]
+                    source_module = source_module.nested_tf_module
+
+        modules: list[Tuple[Module, Dict[TFDefinitionKey, Dict[str, Any]]]] = []
+        if create_graph:
+            modules = []
```

Comment:
would call it `modules_and_definitions_tuple`, maybe you can create a `NamedTuple` also for it
(I know this is how I did it but you are better than me! ;) )